arangod/Pregel/PregelOptions.h

```diff
@@ -0,0 +1,70 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Volmer
+////////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+#include "Basics/ResultT.h"
+#include "Cluster/ClusterTypes.h"
+#include "Inspection/Types.h"
+#include "VocBase/vocbase.h"
+#include "velocypack/Builder.h"
+
+namespace arangodb::pregel {
+
+struct GraphCollectionNames {
+  std::vector<std::string> vertexCollections;
+  std::vector<std::string> edgeCollections;
+};
+
+struct GraphName {
+  std::string graph;
+};
+
+using VertexCollectionID = CollectionID;
+using EdgeCollectionID = CollectionID;
+struct EdgeCollectionRestrictions {
```

Comment:
I wonder whether `EdgeCollectionRestrictions` should be a part of `GraphDataSource`?

Reply:
Good point